src/components/TopBar/TopBar.js

```diff
@@ -308,43 +275,17 @@ const TopBar = ({
             <Toolbar>
                 {logo_clickable}
                 <Box sx={styles.grow}>{children}</Box>
-                {user && withElementsSearch && (
-                    <React.Fragment>
-                        <ElementSearchDialog
-                            open={isDialogSearchOpen}
-                            onClose={() => setDialogSearchOpen(false)}
-                            searchingLabel={searchingLabel}
-                            onSearchTermChange={onSearchTermChange}
-                            onSelectionChange={(element) => {
-                                setDialogSearchOpen(false);
-                                onSelectionChange(element);
-                            }}
-                            elementsFound={elementsFound}
-                            renderElement={renderElement}
-                            searchTermDisabled={searchTermDisabled}
-                            searchTermDisableReason={searchTermDisableReason}
-                        />
-                        <div>
-                            <Button
-                                color="inherit"
-                                onClick={handleClickElementSearch}
-                                disabled={searchDisabled}
-                            >
-                                <SearchIcon />
-                            </Button>
-                        </div>
```

Comment:
Are the other project didn't need the search Dialog ?

Reply:
No, when you [search it in grisuite](https://github.com/search?q=org%3Agridsuite%20withElementsSearch&type=code), you can see only gridstudy-app use it. I keep [ElementSearchDialog](https://github.com/gridsuite/commons-ui/tree/main/src/components/ElementSearchDialog) in commons-ui as it can be reuse easily.